cherab/tools/observers/group/plotting.py

```diff
@@ -0,0 +1,157 @@
+    if ax is None:
+        _, ax = plt.subplots(constrained_layout=True)
+
+    signal = []
+    tick_labels = []
+    for pipeline, observer in zip(pipelines, observers):
+        if isinstance(pipeline, SpectralPowerPipeline0D):
+            spectrum = Spectrum(pipeline.min_wavelength, pipeline.max_wavelength, pipeline.bins)
+            spectrum.samples[:] = pipeline.samples.mean
+            signal.append(spectrum.total())
+        else:
+            signal.append(pipeline.value.mean)
+
+        if observer.name and len(observer.name):
+            tick_labels.append(observer.name)
+        else:
+            tick_labels.append(group._observers.index(observer))
+
+    if isinstance(pipeline, (SpectralRadiancePipeline0D, RadiancePipeline0D)):
+        ylabel = 'Radiance [W/m^2/str]'
+    else:  # SpectralPowerPipeline0D or PowerPipeline0D
+        ylabel = 'Power [W]'
+
+    ax.bar(list(range(len(signal))), signal, tick_label=tick_labels, label=item)
+
+    if isinstance(item, int):
+        # check if pipelines share the same name
+        if len(set(pipeline.name for pipeline in pipelines)) == 1 and pipelines[0].name and len(pipelines[0].name):
+            ax.set_title('{}: {}'.format(group.name, pipelines[0].name))
+        else:
+            # pipelines have different names or name is not set
+            ax.set_title('{}: pipeline {}'.format(group.name, item))
+    elif isinstance(item, str):
+        ax.set_title('{}: {}'.format(group.name, item))
+
+    ax.set_ylabel(ylabel)
+    ax.set_xlabel('Observer')
+
+    return ax
+
+
+def plot_group_spectra(group, item=0, in_photons=False, ax=None):
+    """
+    Plot the spectra observed by each observer in the group for a given pipeline.
+    
+    :param Observer0DGroup group: Group class with observers hodling data to be plotted.
+    :param str/int item: The index or name of the pipeline. Default: 0.
+    :param bool in_photons: If True, plots the spectrum in photon/s/nm instead of W/nm.
+                            Default is False.
+    :param Axes ax: Existing matplotlib axes.
+    :return Axes ax: Matplotlib axes with plotted spectra
+    """
+    pipelines, observers = select_pipelines(group, item)
+
+    if ax is None:
+        _, ax = plt.subplots(constrained_layout=True)
+    
+    for pipeline, observer in zip(pipelines, observers):
+        spectrum_observed = Spectrum(pipeline.min_wavelength, pipeline.max_wavelength, pipeline.bins)
+        spectrum_observed.samples[:] = pipeline.samples.mean
+        if in_photons:
+            # turn the samples into photon/s
+            spectrum = spectrum_observed.new_spectrum()
+            spectrum.samples[:] = spectrum_observed.to_photons()
+        else:
+            spectrum = spectrum_observed
+
+        if spectrum.samples.size > 1:
+            ax.plot(spectrum.wavelengths, spectrum.samples, label=observer.name)
+        else:
+            ax.plot(spectrum.wavelengths, spectrum.samples, marker='o', ls='none', label=observer.name)
```

Comment:
Sorry, I missed this. I think similarly to `plot_group_total()` we should check if the name of the observer is set, and if not, use its index as a label.
```python
label = observer.name if observer.name and len(observer.name) else group.observers.index(observer)
```